### DISCUSSION TITLE
Working with multiple access tokens

### DISCUSSION THREAD

**Discussion by dkundel** [npm/feedback#178] [CLI] [answered]
Hey there! 

I stumbled upon a situation that confused me and I would love your feedback on if there is something that I'm missing.

I have several npm accounts that I'm working with. I'm always logged into the CLI with my personal account but had to publish something using a different account. For that second account I have an access token stored in my environment (example `$SECOND_NPM_TOKEN`). I tried publishing my package using:

```
NPM_TOKEN=$SECOND_NPM_TOKEN npm publish
```

When that didn't work, I tried to explicitly set the `NPM_TOKEN` value and still didn't succeed. I then logged out of the CLI and tried again and still didn't succeed. The only way I was able to achieve it was staying logged out and creating a project specific `.npmrc` file that looked like this:

```
//registry.npmjs.org/:_authToken=${SECOND_NPM_TOKEN}
```

Only then I was able to publish. Is this a bug or am I doing something wrong? Also would it be possible to have a similar flag to --otp for --auth-token so that I could run `npm publish --auth-token=$SECOND_NPM_TOKEN`?

**Comment by MylesBorins** [comment, marked as answer]
Hey! Sorry for missing this discussion. This seems like something to open an issue about over at https://github.com/npm/cli/issues
